src/Deputy/DistrictLoader.php

```diff
@@ -0,0 +1,107 @@
+    {
+        $this->logger = $logger;
+        $this->doctrine = $doctrine;
+        $this->geometryFactory = $geometryFactory;
+        $this->em = $this->doctrine->getManager();
+        $this->decoder = new Serializer([new ObjectNormalizer()], [new JsonEncoder(), new CsvEncoder()]);
+    }
+
+    public function load(string $file, string $districtsFile, string $countriesFile): void
+    {
+        $districts = $this->decoder->decode(file_get_contents($file), 'csv', [CsvEncoder::DELIMITER_KEY => ';']);
+        $this->geoDistricts = $this->decoder->decode(file_get_contents($districtsFile), 'json')['features'];
+        $this->geoCountries = $this->decoder->decode(file_get_contents($countriesFile), 'json')['features'];
+
+        $this->batchInsertOrUpdateDistricts($districts);
+    }
+
+    public function batchInsertOrUpdateDistricts(array $districts): void
+    {
+        $this->logger->notice(sprintf('%s districts are about to be loaded', count($districts)));
+
+        $i = 0;
+
+        foreach ($districts as $district) {
+            $this->createOrUpdateDistrict($district);
+            $this->em->persist($this->createOrUpdateDistrict($district));
+            $this->em->flush();
+
+            if (0 === ++$i % 1000 || $i === count($districts)) {
+                $this->em->clear();
+                $this->logger->notice("$i districts processed");
+            }
+        }
+    }
+
+    private function createOrUpdateDistrict(array $district): District
+    {
+        if (District::FRANCE === $district['code_pays']) {
+            $json = ['ZA', 'ZB', 'ZC', 'ZD', 'ZS', 'ZM', 'ZX', 'ZW', 'ZP', 'ZN'];
+            $csv = ['971', '972', '973', '974', '975', '976', '977', '986', '987', '988'];
+            $code = str_replace($csv, $json, $district['circo_ID']);
+
+            $geoDistricts = array_filter($this->geoDistricts, function ($geoDistrict) use ($code) {
+                return $geoDistrict['properties']['ID'] === $code;
+            });
+
+            if (0 === count($geoDistricts)) {
+                throw new \RuntimeException("Districts GeoJSON file doesn't contain district with code '$code'");
+            }
+            $geoDistrict = array_shift($geoDistricts);
+
+            $geoShape = $this->geometryFactory->createGeometryFromGeoJson($geoDistrict['geometry']);
+            $countries = [District::FRANCE];
+        } else {
+            $countries = explode(',', str_replace(' ', '', $district['code_pays']));
+            $geoCountries = array_filter($this->geoCountries, function ($country) use ($countries) {
+                return in_array($country['properties']['code'], $countries);
+            });
+
+            if (0 === count($geoCountries)) {
+                throw new \RuntimeException("Countries GeoJSON file doesn't contain countries with codes '$countries'");
+            }
+            $geoShape = $this->geometryFactory->mergeGeoJsonGeometries($geoCountries);
```

Comment:
you make your own multipolygon by merging the polygons of each country?
maybe you can use this [official district geo data](https://public.opendatasoft.com/explore/dataset/circonscriptions-legislatives-2017/export/?dataChart=eyJxdWVyaWVzIjpbeyJjb25maWciOnsiZGF0YXNldCI6ImNpcmNvbnNjcmlwdGlvbnMtbGVnaXNsYXRpdmVzLTIwMTciLCJvcHRpb25zIjp7fX0sImNoYXJ0cyI6W3sidHlwZSI6ImNvbHVtbiIsImZ1bmMiOiJDT1VOVCIsInNjaWVudGlmaWNEaXNwbGF5Ijp0cnVlLCJjb2xvciI6IiMyQzNGNTYifV0sInhBeGlzIjoicmVmIiwibWF4cG9pbnRzIjo1MCwic29ydCI6IiJ9XSwidGltZXNjYWxlIjoiIiwiZGlzcGxheUxlZ2VuZCI6dHJ1ZX0%3D) file (but only for France)